frontend/src/templates/Field/Date/DateField.tsx

```diff
@@ -34,23 +35,39 @@ export const DateField = ({
   const isDateUnavailable = useCallback(
     (date: Date) => {
       const { selectedDateValidation } = schema.dateValidation
+      const selectedInvalidDays = schema.invalidDays ?? []
+
       // All dates available.
-      if (!selectedDateValidation) return false
+      if (!selectedDateValidation && selectedInvalidDays.length === 0)
+        return false
+
+      let isDateUnavailable = false
+      let isDayUnavailable = false
 
       switch (selectedDateValidation) {
         case DateSelectedValidation.NoPast:
-          return isDateBeforeToday(date)
+          isDateUnavailable = isDateBeforeToday(date)
+          break
         case DateSelectedValidation.NoFuture:
-          return isDateAfterToday(date)
+          isDateUnavailable = isDateAfterToday(date)
+          break
         case DateSelectedValidation.Custom: {
           const { customMinDate, customMaxDate } = schema.dateValidation
-          return isDateOutOfRange(date, customMinDate, customMaxDate)
+          isDateUnavailable = isDateOutOfRange(
+            date,
+            customMinDate,
+            customMaxDate,
+          )
+          break
         }
         default:
-          return false
+          isDateUnavailable = false
       }
+
+      isDayUnavailable = isDateAnInvalidDay(date, selectedInvalidDays)
+      return isDateUnavailable || isDayUnavailable
```

Comment:
perf: should we return first if `isDateUnavailable` is true so it is (ever so slightly) more efficient?

Reply:
Yes we could return first if `isDateUnavailable` is true.